.github/workflows/check-pr.yaml

```diff
@@ -0,0 +1,103 @@
+name: Build and test
+
+on:
+  pull_request:
+
+jobs:
+  start-vm:
+    if: github.event.pull_request.head.repo.full_name == github.repository
+    name: Start VM
+    runs-on: ubuntu-latest
+    outputs:
+      label: ${{ steps.start-yc-runner.outputs.label }}
+      instance-id: ${{ steps.start-yc-runner.outputs.instance-id }}
+    steps:
+      - name: Start YC runner
+        id: start-yc-runner
+        uses: yc-actions/yc-github-runner@v1
+        with:
+          mode: start
+          yc-sa-json-credentials: ${{ secrets.YC_SA_JSON_CREDENTIALS }}
+          github-token: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
+          folder-id: ${{ vars.YC_FOLDER_ID }}
+          image-id: ${{ vars.YC_DEFAULT_IMAGE_ID }}
+          zone-id: ${{ vars.YC_ZONE_ID }}
+          subnet-id: ${{ vars.YC_SUBNET_ID }}
+          cores: 4
+          memory: 8GB
+          core-fraction: 100
+          disk-type: network-ssd-nonreplicated
+          disk-size: 93GB
+
+  checkout:
+    name: Checkout sources
+    needs: start-vm
+    runs-on: ${{ needs.start-vm.outputs.label }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+  check:
```

Comment:
Let's move this job to a several file as in ytsaurus repo: https://github.com/ytsaurus/ytsaurus/blob/main/.github/workflows/cpp.yaml#L126

Reply:
You mean to reuse the step to avoid duplication, correct?